apps/oauth2/lib/Db/AccessTokenMapper.php

```
@@ -40,27 +41,28 @@ public function __construct(IDBConnection $db) {
 	 * @return AccessToken
 	 * @throws AccessTokenNotFoundException
 	 */
-	public function getByCode($code) {
+	public function getByCode(string $code): AccessToken {
 		$qb = $this->db->getQueryBuilder();
 		$qb
 			->select('*')
 			->from($this->tableName)
 			->where($qb->expr()->eq('hashed_code', $qb->createNamedParameter(hash('sha512', $code))));
-		$result = $qb->execute();
-		$row = $result->fetch();
-		$result->closeCursor();
-		if($row === false) {
-			throw new AccessTokenNotFoundException();
+
+		try {
+			$token = $this->findEntity($qb);
+		} catch (\Exception $e) {
```

Please catch specific exceptions (does not exist, multiple objects returned) instead of the generic one, otherwise this will also swallow syntax errors, db connection issues and so on :wink:

---

Meh but then I ahve 2 catch statements... meh.... but yeah you are right

---

#12887

---

https://wiki.php.net/rfc/multiple-catch for php7.1+ and maybe another argument to drop php7.0 from nextcloud 16 :thinking: